src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteDocumentModelCreater.cs

```diff
@@ -3,13 +3,197 @@
+        {
+            var contentsMetadata = new Dictionary<string, object>();
+            List<OPathSegment> OPathSegments;
+            foreach (var content in contents)
+            {
+                try
+                {
+                    OPathSegments = OverwriteUtility.ParseOPath(content.PropertyName);
+                }
+                catch (ArgumentException ex)
+                {
+                    throw new MarkdownFragmentsException(ex.Message, content.PropertyNameSource.Line, ex);
+                }
+
+                AppendNewObject(OPathSegments, content.PropertyNameSource, content.PropertyValue, contentsMetadata);
+            }
+
+            return contentsMetadata;
+        }
+
+        internal static Dictionary<string, object> MergeYamlCodeMetadataWithContentsMetadata(Dictionary<string, object> yamlCodeMetadata, Dictionary<string, object> contentsMetadata)
+        {
+            var metadata = yamlCodeMetadata.Concat(contentsMetadata).GroupBy(p => p.Key);
+            var metadatasWithSameOPath = from meta in metadata
+                where meta.Count() > 1
+                select meta;
+            foreach (var meta in metadatasWithSameOPath)
+            {
+                Logger.LogWarning(
+                    $"There are two duplicate OPaths `{meta.Key}` in yaml code block and contents block, the item in yaml code block will be overwritten by contents block item",
+                    code: WarningCodes.Overwrite.DuplicateOPaths);
+            }
+
+            return metadata.ToDictionary(g => g.Key, g => g.Last().Value);
+        }
+
+        private static void AppendNewObject(List<OPathSegment> OPathSegments, Block codeHeaderBlock, List<Block> propertyValue, Dictionary<string, object> contentsMetadata)
+        {
+            FindOrCreateObject(contentsMetadata, codeHeaderBlock, OPathSegments, propertyValue,
+                OPathSegments.Select(o => o.OriginalSegmentString).Aggregate((a, b) => a + "/" + b));
+        }
+
+        private static void FindOrCreateObject(Dictionary<string, object> currentObject, Block codeHeaderBlock, List<OPathSegment> leftOPathSegments, List<Block> propertyValue, string originalOPathString)
+        {
+            var segment = leftOPathSegments[0];
+            if (leftOPathSegments.Count == 1)
+            {
+                CreateCoreObject(segment, codeHeaderBlock, currentObject, propertyValue, originalOPathString);
+                return;
+            }
+
+            Dictionary<string, object> nextObject;
+            if (currentObject.TryGetValue(segment.SegmentName, out object childObject))
+            {
+                if (string.IsNullOrEmpty(segment.Key))
+                {
+                    nextObject = childObject as Dictionary<string, object>;
+                    if (nextObject != null)
+                    {
+                        FindOrCreateObject(nextObject, codeHeaderBlock, leftOPathSegments.Skip(1).ToList(), propertyValue, originalOPathString);
```

Comment:
>leftOPathSegments.Skip(1).ToList(), [](start = 72, length = 35)

always passing the original list, and the current index. then the temp lists can be avoided.